packages/beacon-node/test/unit/execution/engine/utils.test.ts

```diff
@@ -0,0 +1,191 @@
+import {expect} from "chai";
+import {ErrorAborted} from "@lodestar/utils";
+import {ExecutePayloadStatus, ExecutionEngineState} from "../../../../src/execution/index.js";
+import {
+  HTTP_CONNECTION_ERROR_CODES,
+  HTTP_FATAL_ERROR_CODES,
+  getExecutionEngineState,
+} from "../../../../src/execution/engine/utils.js";
+import {QueueError, QueueErrorCode} from "../../../../src/util/queue/errors.js";
+import {ErrorJsonRpcResponse, HttpRpcError} from "../../../../src/eth1/provider/jsonRpcHttpClient.js";
+
+describe("execution/engine/utils", () => {
+  describe("getExecutionEngineState", () => {
+    const testCasesPayload: Record<
+      ExecutePayloadStatus,
+      [oldState: ExecutionEngineState, newState: ExecutionEngineState][]
```

Comment:
In test cases, instead of calling it old and new, wouldn't it be clearer to call it from and to?

```suggestion
      [fromState: ExecutionEngineState, toState: ExecutionEngineState][]
```

Both is probably fine

Reply:
The `oldState` | `newState` is used across other places in code. So kept it as it.